### DISCUSSION TITLE
Missing data

### DISCUSSION THREAD

**Discussion by tom-f-oconnell** [voxelmorph/voxelmorph#352] [Ideas]
I was wondering whether you have any ideas about how this library might best be modified to handle missing data, or whether it would be well suited to such a modification (in general / in the specific application I have in mind).

I have some data where I've done two types of two-photon recordings within the same animals:
1. "Functional" recordings of ~5 planes (roughly 100um x 100um) spaced ~12um apart. The goal of these recordings are to sample quickly enough (~1Hz) to get meaningful functional activity within the sampled planes. Many of the structures of interest extend in Z by frequently >= the 12um spacing between the planes.
2. A higher resolution "anatomical" recording with more isotropic imaging parameters (Z-step size of 0.5um), done over a larger total Z-extent (~100um), to assist in segmenting some of the structures in the functional data and hopefully also to provide a bit of an in-animal atlas to be registered to.

I'd like to register the functional data to the anatomical data, either in a slice-to-volume kind of framing, or ideally in a way that still treats the functional data as a volume, just one missing 12um of data between each of the slices. Would it be feasible / sensible to modify voxelmorph for this type of application?

**Comment by adalca** [comment]
@tom-f-oconnell If I understand you correctly, I think this is quite doable, and similar in spirit to @katiemlewis's [CHIL2020 paper](https://arxiv.org/pdf/1812.06932v2.pdf) ? in short, you could give both volumes at the highest resolution (with black or linear-interpolation in the missing areas), and maintains binary 'masks' of what was actually acquired. You can use these masks to guide your loss so as to penalize only acquired data alignment. 

Does that make sense?

**Reply by tom-f-oconnell** [reply]
I think so, thanks. I'll give it a try. The paper does seem like essentially what I was looking for.

**Reply by tom-f-oconnell** [reply]
Where is the code for the paper you referenced? The paper says it is at https://voxelmorph.mit.edu, which redirects to this github repo. The only commit I see that @katiemlewis added is https://github.com/voxelmorph/voxelmorph/commit/7455286f03f5d45caa1e6, which seems to have some of the relevant code, but it would be nice to see a working example, if one exists. Thanks again.

**Reply by tom-f-oconnell** [reply]
Also it appears that the code from Katie's commit has since been deleted from the repo, or at least I haven't found where it has been moved to.

**Reply by adalca** [reply]
@tom-f-oconnell  The code should all be available still @ https://github.com/voxelmorph/voxelmorph/tree/legacy

We don't have a tutorial ready, but @katiemlewis has a schematic that I think is informative, and I have a colab I could edit for you.

**Reply by adalca** [reply]
added a quick [example](https://colab.research.google.com/drive/1m9of9yBvlMie5vBk0WlfCHbHUUsCuKIL?usp=sharing)

**Reply by katiemlewis** [reply]
Yes, I've attached a schematic here for SparseVM. Let me know if there are any questions!
[sparsevm_schematic.pdf](https://github.com/voxelmorph/voxelmorph/files/7197429/sparsevm_schematic.pdf)
